SingularityService/src/main/java/com/hubspot/singularity/resources/AgentResource.java

```diff
@@ -173,6 +173,47 @@ public void decommissionAgent(
     );
   }
 
+  @POST
+  @Path("/agent/{agentId}/decommission/extend")
+  @Operation(summary = "Extend decommission time of a specific active agent")
+  @Consumes({ MediaType.APPLICATION_JSON })
+  public Response extendDecommissionAgent(
+    @Context HttpServletRequest requestContext,
+    @Parameter(hidden = true) @Auth SingularityUser user,
+    @Parameter(required = true, description = "Active agentId") @PathParam(
+      "agentId"
+    ) String agentId,
+    @RequestBody(
+      description = "Settings related to changing the state of a agent"
+    ) SingularityMachineChangeRequest changeRequest
+  ) {
+    return maybeProxyToLeader(
+      requestContext,
+      Response.class,
+      changeRequest,
+      () -> {
+        extendDecommissionAgent(user, agentId, changeRequest);
+        return Response.ok().build();
+      }
+    );
+  }
+
+  public void extendDecommissionAgent(
+    SingularityUser user,
+    String agentId,
+    SingularityMachineChangeRequest changeRequest
+  ) {
+    authorizationHelper.checkAdminAuthorization(user);
```

Comment:
Someone could effectively reactivate or permanently decom an agent with this update, so we should also check `validator.checkActionEnabled(action)` for actions `SingularityAction.ACTIVATE_AGENT` and `SingularityAction.DECOMMISSION_AGENT` just to be safe here.

Reply:
Ok, I can do that. The validator is private to AbstractMachineResource, do you think I should change it to protected or move this function to  AbstractMachineResource?

Reply:
I'd follow whatever pattern the base decom method uses (I think move into AbstractMachineResource)